src/gt4py/backend/module_generator.py

```diff
@@ -136,7 +138,11 @@ def make_args_data_from_gtir(pipeline: GtirPipeline, legacy=False) -> ModuleData
     )
 
     read_fields: Set[str] = set()
-    for expr in node.iter_tree().if_isinstance(gtir.ParAssignStmt).getattr("right"):
+    read_exprs = itertools.chain(
+        node.iter_tree().if_isinstance(gtir.ParAssignStmt).getattr("right"),
+        node.iter_tree().if_isinstance(gtir.FieldIfStmt).getattr("cond"),
+    )
+    for expr in read_exprs:
         read_fields |= expr.iter_tree().if_isinstance(gtir.FieldAccess).getattr("name").to_set()
```

Comment:
@egparedes This is the fix I was referring to. It now looks for `FieldAccess` nodes inside masks as well.